nameko_eventlog_dispatcher/eventlog_dispatcher.py

```diff
@@ -0,0 +1,124 @@
+import logging
+from copy import deepcopy
+from datetime import datetime, timezone
+
+from kombu.common import maybe_declare
+from nameko.amqp import get_connection, verify_amqp_uri
+from nameko.events import EventDispatcher
+from nameko.rpc import Rpc
+from nameko.standalone.events import event_dispatcher, get_event_exchange
+from nameko.web.handlers import HttpRequestHandler
+
+
+log = logging.getLogger(__name__)
+
+
+class EventLogDispatcher(EventDispatcher):
+
+    """Dispatcher of event logs reusing the Nameko `EventDispatcher`.
+
+    When the auto capture feature is enabled, it dispatches an event
+    each time an entrypoint is fired.
+
+    Also provides the ability to manually dispatch event logs,
+    optionally providing related event data.
+
+    On both cases, relevant worker context information is added to the
+    event data.
+    """
+
+    ENTRYPOINT_FIRED = 'entrypoint_fired'
```

Comment:
`EVENT_TYPE` is used to dispatch and handle all the log events and `ENTRYPOINT_FIRED` is just the type of event added to the event meta information.

These two constants would probably be fine in most of the cases. However, I made them class attributes to allow users to override them:

```python
from nameko.messaging import Consumer
from nameko.rpc import Rpc
from nameko.web.handlers import HttpRequestHandler
from nameko_eventlog_dispatcher import EventLogDispatcher


class MyEventLogDispatcher(EventLogDispatcher):

    ENTRYPOINT_FIRED = 'my_entrypoint_fired'
    GENERIC_EVENT_TYPE = 'my_log_event'
    ENTRYPOINT_TYPES_TO_LOG = (Rpc, HttpRequestHandler, Consumer)
```

Then, they could use the new `MyEventLogDispatcher` class. However, your suggestion could still be valid and there could be some value in adding them to the config (and keep the default). I'd like to move this discussion to the next PR I'm preparing if you are happy with it, since I'd like to publish the first version of the library compatible with the way we use it at the moment.